packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/mira/MiraController.java

```diff
@@ -76,36 +78,59 @@ public ResponseEntity<Model> convertAndCreateModel(
 			@RequestBody final ModelConversionRequest conversionRequest) {
 
 		try {
-			TaskRequest req = new TaskRequest();
-			req.setInput(conversionRequest.getModelContent().getBytes());
 
-			if (endsWith(conversionRequest.getModelName(), List.of(".mdl"))) {
+			final Optional<Artifact> artifact = artifactService.getAsset(conversionRequest.artifactId);
```

Comment:
I just want to confirm that this is intentional to use `Artifact` here?

Reply:
Yes, we want to use them for saving user uploaded files that are not part of our java model, like AMR.